### DISCUSSION TITLE
Allow for importing 3d objects (e.g. FBX, GLTF, Collada) as individual objects instead of scenes

### DISCUSSION THREAD

**Discussion by Jonne-G** [godotengine/godot-proposals#6437] [Editor]
Recently I have switched development to Godot, I come from a background of working with Unreal and proprietary game engines.

All game engines besides Godot share the same behaviour when importing a 3d object file,
the file is imported as a complete object even if it is made of multiple sub-objects.

Adding this as an opt-in feature for Godot would (in my opinion) streamline the importing of many 3D objects as it would skip the step of needing to open up the newly imported file as a scene and then extract the desired objects.

Additionally this would improve the experience for developers familiar with other game engines, but new to Godot.

**Comment by Calinou** [comment]
This feature is present in Godot 3.x as the **Instanced Sub-Scenes** import option, but it wasn't reimplemented in the new 3D import system in Godot 4.0: https://github.com/godotengine/godot/issues/56312

In Godot 4, you have an option to select individual meshes in the [Advanced Import Settings dialog](https://docs.godotengine.org/en/latest/tutorials/assets_pipeline/importing_scenes.html#using-the-advanced-import-settings-dialog) then save them to mesh resource files. This saves a Mesh resource, not a scene containing a MeshInstance3D node. This is more useful for working with particle nodes or MeshLibraries which accept Mesh resources, not MeshInstance3D nodes.

**Reply by Jonne-G** [reply]
I am aware of the option to save out the individual meshes through the import dialogue, but this would break single objects that are made out of multiple pieces. The 3.x feature seems to be what I am looking for, but as you say, they aren't saved as meshes, this would be preferred.
I've added more information in a comment below as to why the "export to mesh resource" feature doesn't work in many cases.

**Comment by Jonne-G** [comment]
An example of this would be this Sedan model (found at: https://kenney.nl/assets/car-kit )
[sedan.zip](https://github.com/godotengine/godot-proposals/files/10892347/sedan.zip)
which is made of two pieces:

- sedan_Mesh body
- sedan_Mesh wheel_frontleft

How the file gets imported into godot:
![image](https://user-images.githubusercontent.com/60442130/222974514-7084b07f-e45f-4aca-9db9-b80fdef759a5.png)

Godot imports this as a scene which prevents it from being used in a meshlibrary for instance.
Exporting these pieces individually would make using the car-model as one quite difficult
